argoverse/utils/sim2.py

```diff
@@ -0,0 +1,106 @@
+"""
+Utility for 2d rigid body transformations with scaling.
+
+Refs:
+    http://ethaneade.com/lie_groups.pdf
+    https://github.com/borglab/gtsam/blob/develop/gtsam_unstable/geometry/Similarity3.h
+"""
+
+from typing import Union
+
+import numpy as np
+
+from argoverse.utils.helpers import assert_np_array_shape
+
+
+class Sim2:
+    """ Implements the Similarity(2) class."""
+
+    def __init__(self, R: np.ndarray, t: np.ndarray, s: Union[int, float]) -> None:
+        """Initialize from rotation R, translation t, and scale s.
+
+        Args:
+            R: array of shape (2x2) representing 2d rotation matrix
+            t: array of shape (2,) representing 2d translation
+            s: scaling factor
+        """
+        assert_np_array_shape(R, (2, 2))
+        assert_np_array_shape(t, (2,))
+        assert isinstance(s, float) or isinstance(s, int)
+        self.R_ = R.astype(np.float32)
```

Comment:
Should we make this casting behavior more visible to the user? What if they intended to use `float64`?